### PR TITLE
More docs stuff

### PR DESCRIPTION
**Pull request by nstarman**
## PR Checklist

- [ ] Give a detailed description of the PR above.
- [ ] Document changes in the `CHANGES.rst` file. See existing changelog for examples.
- [ ] Add tests, if applicable, to ensure code coverage never decreases.
- [ ] Make sure the docs are up to date, if applicable, particularly the docstrings and RST files in `docs` folder.
- [ ] Ensure linear history by rebasing, when requested by the maintainer.

### CODE REVIEW

File: docs/api/attributes.rst
Comment (ntessore) [inline docs/api/attributes.rst:1]: Not sure about the grouping of attributes here. We already have the components for that. An alphabetical list of all attributes seems easier to navigate.

File: docs/api/cosmology.rst [hunk 1]
```diff
@@ -23,3 +23,24 @@ At a lower level of abstraction, a cosmology consists of individual
    .. autoproperty:: __cosmology_namespace__
    .. autoproperty:: constants
    .. autoproperty:: name
+
+
+Wrappers
+--------
+
+In order to provide a consistent interface, existing cosmology libraries can be
+wrapped to conform to the :doc:`Cosmology-API <../index>`.  The form of these
```
Comment (ntessore) [inline docs/api/cosmology.rst:32]:
```suggestion
wrapped to conform to the :doc:`Cosmology API <../index>`.  The form of these
```

File: docs/api/cosmology.rst [hunk 2]
```diff
@@ -23,3 +23,24 @@ At a lower level of abstraction, a cosmology consists of individual
...
+wrappers is specified here.
+
+Details of using these wrappers is upcoming. You can see the development of the Astropy and CAMB wrappers in the ``cosmology-api`` github repo.
```
Comment (ntessore) [inline docs/api/cosmology.rst:35]:
```suggestion
Details of using these wrappers is upcoming. You can see the development of the Astropy and CAMB wrappers in the ``cosmology.api`` GitHub repo.
```

File: docs/dev/new.rst [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+library must have a top-level module called "constants".
+
+.. code-block:: text
+
+    library/
+
+        constants.py
+
+
+The :doc:`Cosmology-API <../index>` package provides two run-time checkable
```
Comment (ntessore) [inline docs/dev/new.rst:33]:
```suggestion
The :doc:`Cosmology API <../index>` package provides two run-time checkable
```

File: docs/dev/new.rst [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+- :class:`~cosmology.api.CosmologyNamespace`
+- :class:`~cosmology.api.CosmologyConstantsNamespace`)
+
+
+Cosmology Classes
+-----------------
+
+The second part to a conformant library is to have a cosmology class that
+conforms to the Cosmology API. This is a bit more involved, but not too bad. The
+:doc:`Cosmology-API <../index>` package provides many protocols for the various
```
Comment (ntessore) [inline docs/dev/new.rst:46]:
```suggestion
:doc:`Cosmology API <../index>` package provides many protocols for the various
```

File: docs/index.rst [hunk 5]
```diff
@@ -24,6 +36,14 @@ or transfer functions.
    api/protocols
    api/reference
 
+.. toctree::
+   :caption: Dev
```
Comment (ntessore) [inline docs/index.rst:40]:
```suggestion
   :caption: Developers
```

File: docs/quickstart.rst [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+perform functions requiring many attributes and methods of a Cosmology. Of
+course, the protocols can also be used in simple functions. However, the
+lower-level protocols allow for a more precise description of what a function
+uses. For example, a function requiring only :math:`H_0` and
+:math:`\Omega_{m,0}` can use the corresponding low-level protocols
+:class:`~cosmology.api.HasH0` and :class:`~cosmology.api.HasOmegaM0`, instead of
+the all-encompassing :class:`~cosmology.api.StandardCosmology`.
+
+The last level, the "meta" level, is most useful for developers of cosmology
+libraries and is described in the :doc:`dev section </dev/new>`.
```
Comment (ntessore) [inline docs/quickstart.rst:37]:
```suggestion
libraries and is described in the :doc:`developers section </dev/new>`.
```

File: src/cosmology/api/_core.py [hunk 7]
```diff
@@ -28,7 +28,21 @@ def __cosmology_namespace__(self) -> CosmologyNamespace:
 
     @property
     def name(self) -> str | None:
-        """The name of the cosmology instance."""
+        """The name of the cosmology instance.
+
+        This is a human-friendly label for the cosmology instance. It is
+        optional, and may be `None`. If set, it should be a string. The name is
+        useful for identifying the cosmology instance, for example logging the
+        name in an analysis pipeline, presenting helpful errors messages, and
```
Comment (ntessore) [inline src/cosmology/api/_core.py:36]:
```suggestion
        name in an analysis pipeline, presenting helpful error messages, and
```

File: docs/api/attributes.rst
Comment (nstarman) [reply docs/api/attributes.rst:1]: Sure!

File: docs/dev/typing.rst
[the remark this answers is not in the document]
Comment (nstarman) [reply docs/dev/typing.rst:1]: Done!

File: docs/api/attributes.rst
Comment (nstarman) [reply docs/api/attributes.rst:1]: Done.

File: docs/dev/wrapping.rst [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+compatible with each other.  The Cosmology API is designed to be a common
+interface and we can easily wrap existing codes to make them compatible with the
+Cosmology API.  Many of the common cosmology libraries are already wrapped and
+available in the `cosmology.compat
+<https://github.com/cosmology-api/cosmology-compat>`_ module.
+
+
+If there is an unsopported library that you would like to wrap, the Cosmology
+API is designed to be as simple as possible.  Essentially, you need to create an
+object that implements the :class:`cosmology.api.CosmologyWrapper` interface, as
```
Comment (ntessore) [inline docs/dev/wrapping.rst:16]: The `cosmology.api.CosmologyWrapper` class isn't currently documented anywhere. That's not a problem, I just want to point it out for a potential follow-up issue.

File: docs/dev/new.rst [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+        constants.py
+
+
+The :doc:`Cosmology API <../index>` package provides two run-time checkable
+Protocols that can be used for static and run-time checks that the library and the
+constants module conform to the API. See the protocols for details.
+
+- :class:`~cosmology.api.CosmologyNamespace`
+- :class:`~cosmology.api.CosmologyConstantsNamespace`)
```
Comment (ntessore) [inline docs/dev/new.rst:38]: There's an extraneous `)` at the end of this line. Tagging for possible follow-up.

**Review by ntessore — APPROVED** [review]

File: docs/dev/wrapping.rst (hunk 8 above)
Comment (nstarman) [reply docs/dev/wrapping.rst:16]: Yeah, we need a section in both the user docs and dev docs about it.

File: docs/dev/new.rst (hunk 9 above)
Comment (nstarman) [reply docs/dev/new.rst:38]: Made a followup issues.

**Pull request merged**